src/traces/parcoords/attributes.js

```diff
@@ -0,0 +1,188 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var colorAttributes = require('../../components/colorscale/color_attributes');
+var colorbarAttrs = require('../../components/colorbar/attributes');
+var axesAttrs = require('../../plots/cartesian/layout_attributes');
+
+var extendFlat = require('../../lib/extend').extendFlat;
+
+module.exports = {
+
+    id: {
```

Comment:
No, we don't need this. `cleanData` which gets called early on in `Plotly.plot` already adds `uid` to all traces [here](https://github.com/plotly/plotly.js/blob/74dbbc4277cf9f3f2ee74d6d9264aac4c201261a/src/plot_api/helpers.js#L220-L223)